### PR TITLE
Remove outdated technical overview

### PR DESCRIPTION
**Pull request by saulmc**
This section of the README reflects our thinking prior to the decision to adopt MLS.

### CODE REVIEW

**Review by neekolas — APPROVED** [review]

**Comment by neekolas** [comment]
Thanks for catching this

File: README.md [hunk 1]
```diff
@@ -56,49 +56,3 @@ $ docker build . -t libxmtp:1
...
-
-- All of their past messages might not remain secure.
-- Future messages might not remain secure.
-
-**How**: Specifically, it uses the Rust version of [Olm](https://gitlab.matrix.org/matrix-org/olm/-/blob/master/docs/olm.md), which is Matrix protocol’s implementation of double ratchet. To learn more about the double ratchet algorithm and how sessions and session keys work, see [The Double Ratchet Algorithm](https://signal.org/docs/specifications/doubleratchet/).
-
-> **Tip**  
-> Use the [example CLI](https://github.com/xmtp/libxmtp/blob/main/examples/cli/README.md) to try out sending XMTP v3-alpha double ratchet messages on the XMTP `dev` network.
-
-### Installation key bundles
```
Comment (neekolas) [inline README.md:86]: This section is basically still accurate, but we cover it in the new credentials doc anyways.

**Comment by saulmc** [comment]
This is what I get for trying to slip in an easy PR...

CleanShot 2024-01-16 at 11 27 07@2x

**Comment by neekolas** [comment]
I don't know what the deal is there @saulmc. The last PR to `main` worked. I wonder if something is off with our build caching in CI?

I'm just gonna admin-merge this, since there's no way your change is related to the issue.

**Pull request merged**